client/trino-jdbc/src/test/java/io/trino/jdbc/TestTrinoDatabaseMetaData.java

```diff
@@ -1254,6 +1319,38 @@ public void testGetTablesMetadataCalls()
                     list(),
                     ImmutableMultiset.of());
         }
+
+        try (Connection connection = createConnectionWithNullCatalogMeansCurrent()) {
+            assertMetadataCalls(
+                    connection,
+                    readMetaData(
+                            databaseMetaData -> databaseMetaData.getTables(null, null, null, new String[0]),
+                            list("TABLE_CAT", "TABLE_SCHEM", "TABLE_NAME", "TABLE_TYPE")),
+                    list(),
+                    ImmutableMultiset.of());
+
+            // set a different catalog to check if current catalog is used
+            connection.setCatalog(COUNTING_CATALOG);
+
+            assertMetadataCalls(
+                    connection,
+                    readMetaData(
+                            databaseMetaData -> databaseMetaData.getTables(null, null, null, new String[0]),
+                            list("TABLE_CAT", "TABLE_SCHEM", "TABLE_NAME", "TABLE_TYPE")),
+                    list(),
+                    ImmutableMultiset.of());
+
+            // change the catalog back using a statement on the connection
+            connection.createStatement().execute(String.format("USE %s.%s", "blackhole", "blackhole"));
+
+            assertMetadataCalls(
+                    connection,
+                    readMetaData(
+                            databaseMetaData -> databaseMetaData.getTables(null, null, null, new String[0]),
+                            list("TABLE_CAT", "TABLE_SCHEM", "TABLE_NAME", "TABLE_TYPE")),
+                    list(),
+                    ImmutableMultiset.of());
+        }
```

Comment:
what do these assertions test? They appear the same to me?

Same for the `getColumns` ones below.